ister_gui.py

```diff
@@ -644,6 +672,255 @@ def validate_condition(self):
+        output = subprocess.check_output(cmd).decode('utf-8')
+        for line in output.split('\n'):
+            if '/' in line:
+                return line.split()[0]
+
+        return ''
+
+    def _mount_host_disk(self, config):
+        """Search for and mount the host os disk"""
+        root_part = ''
+        boot_part = ''
+        prefix = 'ister-{}-'.format(str(config["Version"]))
+        self.target_dir = tempfile.mkdtemp(prefix=prefix)
+        disks = get_list_of_disks()
+        # search disks for root and boot partitions
+        for disk in disks:
+            if disk in self.current or root_part:
+                continue
+            part_info = get_disk_info('/dev/{}'.format(disk))
+            root_part = self._get_part(part_info,
+                                       'Linux root',
+                                       self.target_dir)
+            # only look for a boot partition on the same disk as the root
+            # partition and only if a root partition is found
+            if root_part:
+                boot_part = self._get_part(part_info,
+                                           'EFI System',
+                                           '{}/boot'.format(self.target_dir))
+
+        # error handling
+        if not root_part:
+            self.error = True
+            Alert('Error!', 'Unable to find host OS in: {}'
+                            .format(', '.join(disks))).do_alert()
+            return (root_part, boot_part)
+
+        if not boot_part:
+            Alert('Information', 'Unable to find boot partition for {}'
+                                 .format(root_part)).do_alert()
+
+        return (root_part, boot_part)
+
+
+    def _get_part(self, part_info, pattern, target):
+        """
+        Find partition specified by pattern and mount it at target.
+        This needs to return an empty string if it fails to find or mount the
+        pattern so that it doesn't later break subprocess calls.
+        """
+        for part in part_info['partitions']:
+            if part.get('type') and pattern in part['type']:
+                part_found = part['name']
+                try:
+                    # try to mount partition
+                    mount_res = subprocess.call(['mount',
+                                                 part_found,
+                                                 target])
+                    # only mount a linux root partition if it is Clear Linux OS
+                    if 'Linux root' in pattern:
+                        osf = '{}/usr/lib/os-release'.format(self.target_dir)
+                        with open(osf, 'r') as os_release:
+                            if 'clear-linux-os' in os_release.read():
+                                return part_found
+                            else:
+                                # maybe there is another root part on this disk
+                                continue
```

Comment:
Need to umount for all the mounted partitions that weren't a Clear Linux rootfs.